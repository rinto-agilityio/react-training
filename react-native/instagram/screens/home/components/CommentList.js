import React from "react";
import { View, Text, StyleSheet } from "react-native";

export default class CommentList extends React.Component {
  render() {
    const { comments } = this.props;

    return (
      <View>
        {comments.map(comment => {
          return (
            <Text key={comment.id}>
              <Text style={styles.username}>{comment.owner.username}</Text>
              {": " + comment.text}
            </Text>
          );
        })}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  username: {
    fontWeight: "bold",
    marginRight: 10,
    paddingRight: 10
  }
});
