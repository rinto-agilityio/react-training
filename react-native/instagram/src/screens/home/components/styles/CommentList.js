import { StyleSheet } from 'react-native'

export const styles = StyleSheet.create({
  comment: {
    marginTop: 5
  }
})
