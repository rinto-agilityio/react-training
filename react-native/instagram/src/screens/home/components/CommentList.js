import React from 'react';
import { View, Text } from 'react-native';

import { styles } from './styles/CommentList';
import CommonStyles from '../../../theme/common';
export default class CommentList extends React.Component {
  render() {
    const { comments } = this.props;

    return (
      <View>
        {comments.map(comment => {
          return (
            <Text key={comment.id} style={styles.comment}>
              <Text style={CommonStyles.textBold}>
                {comment.owner.username}
              </Text>
              {': ' + comment.text}
            </Text>
          );
        })}
      </View>
    );
  }
}
