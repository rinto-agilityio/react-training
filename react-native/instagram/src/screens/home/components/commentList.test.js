import { comments } from '../../../test/__mocks__/sample-data'
import CommentList from './CommentList'

describe('CommentList component', () => {
  it('CommentList renders correctly', () => {
    const treeDOM = renderer
      .create(<CommentList comments={comments} />)
      .toJSON()

    expect(treeDOM).toMatchSnapshot()
  })
})
