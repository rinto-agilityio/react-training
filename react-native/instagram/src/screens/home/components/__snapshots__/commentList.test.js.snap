// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentList component CommentList renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginTop": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      instagram
    </Text>
    : Awesome photo
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginTop": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      phorgasm
    </Text>
    : Woow beautiful color.
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginTop": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      instagram
    </Text>
    : This is art
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginTop": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      intercontinentaldanang
    </Text>
    : Wow this is a fresh concept
  </Text>
</View>
`;
